website/docs/r/nsxt_firewall.html.markdown

```diff
@@ -0,0 +1,117 @@
+## Example Usage 1 (Single rule to allow all IPv4 traffic from anywhere to anywhere)
+```hcl
+resource "vcd_nsxt_firewall" "testing" {
+  org  = "my-org"
+  vdc  = "my-nsxt-vdc"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  rule {
+    name        = "allow all IPv4 traffic"
+    direction   = "IN_OUT"
+    ip_protocol = "IPV4"
+  }
+}
+```
+
+## Example Usage 2 (Multiple firewall rules - order matters)
+```hcl
+resource "vcd_nsxt_firewall" "testing" {
+  org  = "my-org"
+  vdc  = "my-nsxt-vdc"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  # Rule #1 - Allows in IPv4 traffic from security group `vcd_nsxt_security_group.group1.id`
+  rule {
+    name        = "first rule"
+    direction   = "IN"
+    ip_protocol = "IPV4"
+    source_ids  = [vcd_nsxt_security_group.frontend.id]
+  }
+
+  # Rule #2 - Drops and logs all outgoing IPv6 traffic to `vcd_nsxt_security_group.group.2.id`
+  rule {
+    name            = "drop IPv6 with destination to security group 2"
+    direction       = "OUT"
+    ip_protocol     = "IPV6"
+    destination_ids = [vcd_nsxt_security_group.group2.id]
+    action          = "DROP"
+    logging         = true
+  }
+  
+  # Rule #3 - Allows IPv4 and IPv6 traffic of 2 Application Port Profiles in both directions:
+  # from vcd_nsxt_security_group.group.1.id to all list of security groups vcd_nsxt_security_group.group.*.id
+  # from list of security groups vcd_nsxt_security_group.group.*.id to vcd_nsxt_security_group.group.1.id
+  rule {
+    name            = "test_rule-3"
+    direction       = "IN_OUT"
+    ip_protocol     = "IPV4_IPV6"
+    source_ids      = [vcd_nsxt_security_group.group.1.id]
+    destination_ids = vcd_nsxt_security_group.group.*.id
+    app_port_profile_ids = [app_port_profile_ids = [data.vcd_nsxt_app_port_profile.ssh.id, vcd_nsxt_app_port_profile.custom-app.id]
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+  when connected as sysadmin working across different organisations.
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level.
+* `edge_gateway_id` - (Required) The ID of the edge gateway (NSX-T only). Can be looked up using
+  `vcd_nsxt_edgegateway` datasource
+* `rule` (Required) One or more blocks with [Firewall Rule](#firewall-rule) definitions
+
+<a id="firewall-rule"></a>
+## Firwall Rule
+
+Each Firewall Rule contains following attributes:
+
+* `name` - (Required) Explanatory name for firewall rule (uniqueness not enforced)
+* `direction` - (Required) One of `IN`, `OUT`, or `IN_OUT`
+* `ip_protocol` - (Required) One of `IPV4`,  `IPV6`, or `IPV4_IPV6`
+* `enabled` - (Optional) Defines if the rule is enabled (default `true`)
+* `logging` - (Optional) Defines if logging for this rule is enabled (default `false`)
+* `action` - (Optional) Defines if it should `ALLOW` or `DROP` traffic (default `ALLOW`)
```

Comment:
I was looking at the examples and am hesitant about having a default (ALLOW) here. It's critical field and not seeing the actual value explicitly in HCL, makes it harder/riskier to use, IMHO. What do you and @dataclouder think? How does UI behave?

Reply:
UI sets `Allow` by default as there is this default rule "deny all"  in the end:
![image](https://user-images.githubusercontent.com/15804230/121866347-82ed7f80-cd07-11eb-827c-7f570b2a91b8.png)

I am totally open to requiring it explicitly :) Not an issue maybe it enforces readability when checking without context.

Reply:
Unless there's some corner cases, which I'm not aware of, I vote for explicitly. We are doing that in other places where a binary choice exists in a critical field. To some extent, UI is also explicit in the sense that it is showing the Allow in the list.

Reply:
Implemented this in d17a124